src/test/java/org/jenkinsci/plugins/envinject/EnvInjectJobPropertyTest.java

```diff
@@ -0,0 +1,134 @@
+package org.jenkinsci.plugins.envinject;
+
+import hudson.model.Cause;
+import hudson.model.FreeStyleBuild;
+import hudson.model.FreeStyleProject;
+import hudson.model.ParametersAction;
+import hudson.model.Result;
+import hudson.model.StringParameterValue;
+import hudson.model.TaskListener;
+import hudson.model.queue.QueueTaskFuture;
+import hudson.tasks.Shell;
+import org.junit.Rule;
+import org.jvnet.hudson.test.JenkinsRule;
+
+import static org.junit.Assert.*;
+import org.junit.Ignore;
+import org.junit.Test;
+import org.jvnet.hudson.test.CaptureEnvironmentBuilder;
+
+/**
+ * Tests for {@link EnvInjectJobProperty}
+ * @author Oleg Nenashev
+ */
+public class EnvInjectJobPropertyTest {
+    
+    @Rule
+    public JenkinsRule j = new JenkinsRule();
+
+    @Test
+    public void testVariableInjection() throws Exception {   
+        FreeStyleProject project = j.createFreeStyleProject();
+        
+        EnvInjectJobProperty<FreeStyleProject> prop = new EnvInjectJobProperty<FreeStyleProject>();
+        prop.setInfo(new EnvInjectJobPropertyInfo(null, "FOO=BAR", null, null, null, false));
+        prop.setOn(true);
+        project.addProperty(prop);
+        
+        FreeStyleBuild build = j.buildAndAssertSuccess(project);
+        assertEquals("BAR", build.getEnvironment(TaskListener.NULL).get("FOO"));
+    }
+    
+    @Test
+    public void testVariableInjection_PropertyDisabled() throws Exception {   
+        FreeStyleProject project = j.createFreeStyleProject();
+        
+        EnvInjectJobProperty<FreeStyleProject> prop = new EnvInjectJobProperty<FreeStyleProject>();
+        prop.setInfo(new EnvInjectJobPropertyInfo(null, "FOO=BAR", null, null, null, false));
```

Comment:
what does mean all this nulls and false? Think the only useful thing is var value. So this two lines can be replaced with one method:
```java
public EnvInjectJobProperty prop(String content) {
       EnvInjectJobProperty<FreeStyleProject> prop = new EnvInjectJobProperty<FreeStyleProject>();
       prop.setInfo(new EnvInjectJobPropertyInfo(null, content, null, null, null, false));
}
```

so it simplifies all tests